### ISSUE TITLE
Exception: "surrogates not allowed" when previewing [System / Cache] on Linux Mint 20.2

### ISSUE THREAD

**Issue by markSmurphy** [bleachbit/bleachbit#1239]
When I select `Cache` under `System` and then choose `Preview`, BleachBit scans through ~27000 files with 1 error:

`Exception while running operation 'system': ''utf-8' codec can't encode character '\udce6' in position 68: surrogates not allowed'`

**To Reproduce**
Steps to reproduce the behaviour:
1. Select `Cache` under `System`
2. Click on `Preview`
3. Scroll down to `/home/user/.cache/Microsoft/Edge/IdentityCache/`

The error happens every time.

**Expected behaviour**
The preview should scan through all files without reporting an error

**Screenshots**
The full error is:
```text
Exception while running operation 'system': ''utf-8' codec can't encode character '\udce6' in position 68: surrogates not allowed'
Traceback (most recent call last):
  File "/usr/share/bleachbit/Worker.py", line 347, in run_operations
    for dummy in self.clean_operation(operation):
  File "/usr/share/bleachbit/Worker.py", line 160, in clean_operation
    for ret in self.execute(cmd, '%s.%s' % (operation, option_id)):
  File "/usr/share/bleachbit/Worker.py", line 132, in execute
    self.ui.append_text(line)
  File "/usr/share/bleachbit/GUI.py", line 635, in append_text
    self.textbuffer.insert(__iter, text)
  File "/usr/lib/python3/dist-packages/gi/overrides/Gtk.py", line 765, in insert
    Gtk.TextBuffer.insert(self, iter, text, length)
UnicodeEncodeError: 'utf-8' codec can't encode character '\udce6' in position 68: surrogates not allowed
```
![Screenshot from 2021-07-31 10-40-12](https://user-images.githubusercontent.com/9842107/127736035-b48cf723-8e8d-4c81-9e79-6510a6919b23.png)

**Desktop (please complete the following information):**
From BlechBit's System Information:
```text
BleachBit version 4.4.0
Git revision 5f55050
GTK version 3.24.20
GTK theme = Mint-Y
GTK icon theme = Mint-Y
GTK prefer dark theme = True
SQLite version 3.31.1
local_cleaners_dir = None
locale_dir = /usr/share/locale/
options_dir = /home/mark/.config/bleachbit
personal_cleaners_dir = /home/mark/.config/bleachbit/cleaners
system_cleaners_dir = /usr/share/bleachbit/cleaners
locale.getdefaultlocale = ('en_GB', 'UTF-8')
os.getenv('DESKTOP_SESSION') = cinnamon
os.getenv('LOGNAME') = mark
os.getenv('USER') = mark
os.getenv('SUDO_UID') = None
os.path.expanduser('~') = /home/mark
platform.linux_distribution() = ('LinuxMint', '20.2', 'uma')
platform.platform = Linux-5.4.0-80-generic-x86_64-with-glibc2.29
platform.version = #90-Ubuntu SMP Fri Jul 9 22:49:44 UTC 2021
sys.argv = ['/usr/bin/bleachbit']
sys.executable = /usr/bin/python3
sys.version = 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0]
__file__ = /usr/share/bleachbit/SystemInformation.py
```
![Screenshot from 2021-07-31 10-51-09](https://user-images.githubusercontent.com/9842107/127736093-75b8b6b7-3e43-4274-adfe-aba4997ffc9c.png)

**Additional context**
If I navigate to the parent folder where the error occurs and run `tree -a` (to list all files in the directory structure) I get:
```bash
.
└── e_C2GK9UTC67FSUCIV
    ├── Accounts
    │   └── r_75FR6NURNR2TGPBK.bin
    ├── AT
    │   └── r_75FR6NURNR2TGPBK
    │       ├── c_51FU2JLLEJE759AV.bin
    │       ├── c_Q1JJ95IJFFRLV2IV.bin
    │       └── c_SR6F7RAVM7CLU9QU.bin
    ├── FRT
    │   └── f_DE3B4SVV6JUE37BB.bin
    ├── ID
    │   └── r_75FR6NURNR2TGPBK
    │       ├── c_51FU2JLLEJE759AV.bin
    │       ├── c_Q1JJ95IJFFRLV2IV.bin
    │       └── c_SR6F7RAVM7CLU9QU.bin
    └── RT
        ├── c_51FU2JLLEJE759AV.bin
        ├── c_Q1JJ95IJFFRLV2IV.bin
        └── c_SR6F7RAVM7CLU9QU.bin

8 directories, 11 files
```
This only seems to occur in path `/home/user/.cache/Microsoft/Edge/IdentityCache/` which strongly implies it's something to do with how the Chromium build of Edge writes files. There could well be other packages which write files which could also trip BleachBit up, so I thought it's worth reporting here.
If you're curious, I'm using Edge over Chrome because a fresh install of the latest Chrome version had issues logging in to a Google account secured with MFA, while Edge didn't!!!

**Comment by az0** [comment]
@markSmurphy 
Thank you for the report with all the detail. Which version of Edge? Is it the stable version? How did you install it (e.g., snap, apt)?

**Comment by markSmurphy** [comment]
@az0 Apologies for the tardy response.

I am on the `dev` channel for Edge. At the time of writing this response it's version `95.0.997.1`. This issue has persisted across v94 --> v95 though.

 I installed via adding an additional `apt` repository:
```text
deb [arch=amd64] http://packages.microsoft.com/repos/edge/ stable main
```

**Comment by az0** [comment]
I can't reproduce on Ubuntu 22.04 with the latest BleachBit (git main bc81da0c6eb22a577d5d16599ec68338db0f2944). 

apt source: `deb [arch=amd64] https://packages.microsoft.com/repos/edge/ stable main`

apt install: `sudo apt-install microsoft-edge-beta`

Microsoft Edge version: 114.0.1823.30 (Official build) beta (64-bit) 

I tried

* both the BleachBit GUI and CLI
* both System - Cache (like the original message above)
* all the options under the Microsoft Edge cleaner (added today: bc81da0c6eb22a577d5d16599ec68338db0f2944 )

@markSmurphy  Do you still have this issue?

**Comment by az0** [comment]
No reply since May 27

**Comment by az0** [comment]
This looks like https://github.com/bleachbit/bleachbit/issues/1709 fixed by ac6e8ec08e2f6b4fa6751f59e5b47ddc88083c80, so this was fixed in 6.0.0
